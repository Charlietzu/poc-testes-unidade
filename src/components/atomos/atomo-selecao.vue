<template>
  <v-select
    :items="itens"
    v-model="itemInterno"
    :item-text="textoItem"
    :item-value="valorItem"
    :placeholder="placeholder"
    dense
    outlined
    item-color="primary"
    @change="onChange"
    no-data-text="Os dados não estão disponíveis"
  ></v-select>
</template>

<script lang="ts">
import Vue from "vue";
import Component from "vue-class-component";
import { Prop, Emit, Watch } from "vue-property-decorator";

@Component
export default class AtomoSelecao extends Vue {
  private itemInterno: number | null = null;

  @Watch("itemInterno")
  alterarItem(): void {
    this.itemInterno = this.itemSelecionado;
  }

  created(): void {
    this.itemInterno = this.itemSelecionado;
  }

  @Watch("itemInterno")
  @Emit("alteracaoSelect")
  onChange(valorSelecionado: number): number {
    return valorSelecionado;
  }

  @Prop()
  itemSelecionado!: number | null;

  @Prop()
  itens!: any[];

  @Prop()
  textoItem!: string;

  @Prop()
  valorItem!: number;

  @Prop({ type: String, default: "" })
  placeholder?: string;
}
</script>

<style lang="scss" scoped></style>
