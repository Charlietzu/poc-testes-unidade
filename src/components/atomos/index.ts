import AtomoSelecao from "./atomo-selecao.vue";

export { AtomoSelecao };
