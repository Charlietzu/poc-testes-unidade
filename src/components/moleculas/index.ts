import MoleculaSelecao from './molecula-selecao.vue';

export { MoleculaSelecao };
