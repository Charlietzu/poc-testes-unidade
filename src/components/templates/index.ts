import TemplatePadrao from "./template-padrao.vue";

export { TemplatePadrao };
