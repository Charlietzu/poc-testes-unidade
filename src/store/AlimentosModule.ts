import { Action, Module, Mutation, VuexModule } from "vuex-module-decorators";

export interface Alimento {
  nomeAlimento: string;
  codigoAlimento: number;
}

@Module({ name: "Alimentos" })
export default class AlimentosModule extends VuexModule {
  private _alimentos: Alimento[] = [
/*     {
      nomeAlimento: "Alface",
      codigoAlimento: 0,
    },
    {
      nomeAlimento: "Arroz",
      codigoAlimento: 1,
    },
    {
      nomeAlimento: "Batata",
      codigoAlimento: 2,
    }, */
  ];

  private _alimentoSelecionado: number | null = null;

  get alimentos(): Alimento[] {
    return this._alimentos;
  }

  get alimentoSelecionado(): number | null {
    return this._alimentoSelecionado;
  }

  @Mutation
  private setAlimentos(alimentos: Alimento[]): void {
    this._alimentos = alimentos;
  }

  @Mutation
  private setAlimentoSelecionado(codigoAlimento: number): void {
    this._alimentoSelecionado = codigoAlimento;
  }

  @Action
  atualizarAlimentoSelecionado(codigoAlimento: number): void {
    this.context.commit("setAlimentoSelecionado", codigoAlimento);
  }

  @Action
  obterAlimentos(alimentos: Alimento[]): void {
    this.context.commit("setAlimentos", alimentos);
  }
}
