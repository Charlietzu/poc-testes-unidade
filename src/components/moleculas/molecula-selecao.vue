<template>
  <atomo-selecao
    :itens="alimentos"
    :itemSelecionado="alimentoSelecionado"
    valorItem="codigoAlimento"
    textoItem="nomeAlimento"
    placeholder="Selecione Alimento"
    @alteracaoSelect="alteracaoSelecaoAlimento"
  ></atomo-selecao>
</template>

<script lang="ts">
import Component from "vue-class-component";
import Vue from "vue";
import { AtomoSelecao } from "@/components/atomos";
import { Alimento } from "@/store/AlimentosModule";
import { moduloAlimentos } from "@/store";
@Component({
  components: {
    AtomoSelecao,
  },
})
export default class MoleculaSelecao extends Vue {
  get alimentos(): Alimento[] {
    return moduloAlimentos.alimentos;
  }

  get alimentoSelecionado(): number | null {
    return moduloAlimentos.alimentoSelecionado;
  }

  alteracaoSelecaoAlimento(codigoAlimentoSelecionado: number): void {
    console.log('evento emitido')
    moduloAlimentos.atualizaAlimentoSelecionado(codigoAlimentoSelecionado);
  }
}
</script>
