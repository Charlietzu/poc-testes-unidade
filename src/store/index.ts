import Vue from "vue";
import Vuex from "vuex";
import AlimentosModule from "./AlimentosModule";
import { getModule } from "vuex-module-decorators";

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Alimentos: AlimentosModule,
  },
});

export default store;

const moduloAlimentos = getModule(AlimentosModule, store);

export { moduloAlimentos };
