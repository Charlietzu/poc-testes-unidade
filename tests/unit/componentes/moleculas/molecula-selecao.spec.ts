import { createLocalVue, mount } from "@vue/test-utils";
import { MoleculaSelecao } from "@/components/moleculas";
import Vuetify from "vuetify";
import Vue, { VueConstructor } from "vue";
import { AtomoSelecao } from "@/components/atomos";
import Vuex from "vuex";
import store from "@/store";

const localVue: VueConstructor<Vue> = createLocalVue();
localVue.use(Vuex);
describe("molecula-selecao.vue", () => {
  let vuetify: Vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const montarComponente = (options: Record<string, unknown>) => {
    return mount(MoleculaSelecao, {
      store,
      localVue,
      vuetify,
      ...options,
    });
  };
  it("Renderiza o componente corretamente", () => {
    //Arrange & Act
    const wrapper = montarComponente({});
    //Assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Deve ter o componente AtomoSelecao como filho", () => {
    //Arrange & Act
    const wrapper: any = montarComponente({});

    //Assert
    expect(wrapper.findComponent(AtomoSelecao).exists()).toBe(true);
  });

  it("Deve chamar alteracaoSelecaoAlimento caso AtomoSelecao emita evento alteracaoSelect", async () => {
    //Arrange
    const wrapper: any = montarComponente({});

    //Act
    await wrapper.findComponent(AtomoSelecao).vm.$emit("alteracaoSelect", 2);
    await wrapper.vm.$nextTick();

    //Assert
    expect(wrapper.vm.alimentoSelecionado).toBe(2);
  });
});
