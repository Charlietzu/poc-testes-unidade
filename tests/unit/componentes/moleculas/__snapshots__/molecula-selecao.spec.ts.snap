// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecula-selecao.vue Renderiza o componente corretamente 1`] = `
<div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder v-select">
  <div class="v-input__control">
    <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-5" class="v-input__slot">
      <fieldset aria-hidden="true">
        <legend style="width: 0px;"><span class="notranslate">â€‹</span></legend>
      </fieldset>
      <div class="v-select__slot">
        <div class="v-select__selections"><input id="input-5" placeholder="Selecione Alimento" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
        <div class="v-input__append-inner">
          <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
        </div><input type="hidden">
      </div>
      <div class="v-menu">
        <!---->
      </div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
